runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -1120,6 +1191,36 @@ TR_J9InlinerPolicy::createUnsafePutWithOffset(TR::ResolvedMethodSymbol *calleeSy
    // Preserve null check on the unsafe object
    TR::TransformUtil::separateNullCheck(comp(), callNodeTreeTop, tracer()->debugLevel());
 
+
+   bool conversionNeeded = comp()->fe()->dataTypeForLoadOrStore(type) != type;
+
+   // If we are not able to get javaLangClass it is still inefficient to put direct Access far
+   // So in that case we will generate lowTagCmpTest to branch to indirect access if true
+   TR_OpaqueClassBlock *javaLangClass = comp()->getClassClassPointer(/* isVettedForAOT = */ true);
+   bool needNotLowTagged = javaLangClass != NULL  || conversionNeeded ;
+
+   int length;
+   const char *objTypeSig = unsafeCall->getChild(1)->getSymbolReference()->getTypeSignature(length);
```

Comment:
you need to consider the possible case where getChild(1) returns NULL.  i have seen such java code and dealt with such PMRs of crash.